src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -43,6 +43,7 @@ pub struct BuildConfig {
     pub exec_engine: Option<Arc<Box<ExecEngine>>>,
     pub release: bool,
     pub doc_all: bool,
+    pub target_rustc_args: Option<Vec<String>>,
```

Comment:
I think that it may be best to add this option to a `Profile` instead of the build configuration here (reflecting on this). Right now the custom arguments are passed to *all* crates in the dependency graph, but the command is tasked with only passing the arguments to the top-level command.

By using a `Profile` you can automatically set the extra arguments to an empty list for all profiles generated for dependencies, but the top-level profiles (generated in `cargo_compile.rs`) can have the arguments listed.